source/PlayerInfo.cpp

```diff
@@ -2639,6 +2639,111 @@ void PlayerInfo::RegisterDerivedConditions()
 		return retVal;
 	});
 
+	// The total number of ships the player has.
+	auto &&totalShipsProvider = conditions.GetProviderNamed("total ships");
+	totalShipsProvider.SetGetFunction([this](const string &name) -> int64_t
+	{
+		int64_t retVal = 0;
+		for(const shared_ptr<Ship> &ship : ships)
+			if(!ship->IsDestroyed())
+				++retVal;
+		return retVal;
+	});
+
+	// Conditions to determine what outfits the player owns, with various possible locations to check.
+	// The following condition checks all sources of outfits which are present with the player.
+	// If in orbit, this means checking all ships in-system for installed and in cargo outfits.
+	// If landed, this means checking all landed ships for installed outfits, the pooled cargo hold,
+	// and any planetary storage.
+	auto &&presentOutfitProvider = conditions.GetProviderPrefixed("outfit (present): ");
+	presentOutfitProvider.SetGetFunction([this](const string &name) -> int64_t
+	{
+		int64_t retVal = 0;
+		const Outfit *outfit = GameData::Outfits().Get(name);
```

Comment:
The Get-function should not create the outfit if it doesn't exist yet. (Use `Find()` instead of `Get()`). Also for the functions below.